src/main/java/TechDetectives/HobbyFinder/controllers/HomeController.java

```diff
@@ -0,0 +1,84 @@
+package TechDetectives.HobbyFinder.controllers;
+
+
+import TechDetectives.HobbyFinder.Models.Data.CategoryRepository;
+
+import TechDetectives.HobbyFinder.Models.Category;
+import TechDetectives.HobbyFinder.Models.Data.HobbyRepository;
+import TechDetectives.HobbyFinder.Models.Hobby;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@Controller
+public class HomeController {
+
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    @Autowired
+    private HobbyRepository hobbyRepository;
+
+    @RequestMapping("/")
+    public String index(Model model) {
+        model.addAttribute("hobbies", hobbyRepository.findAll());
+        return "index";
+
+    }
+
+    @GetMapping("add")
+    public String displayAddHobbyForm(Model model) {
+        model.addAttribute("title", "Add Hobby");
+        model.addAttribute(new Hobby());
+
+        List categories = (List<Category>) categoryRepository.findAll();
+        model.addAttribute("categories", categories);
+        return "add";
+    }
+
+
+
+    @PostMapping("add")
```

Comment:
Should this be "hobby/add" instead of "add" to be more specific?

Reply:
Because comment is going to be "comment/add" so that they know which endpoint its sending the "add" to.

Reply:
Do you want me to add "hobby" folder and put "add  thymeleaf " below?